fe/fe-core/src/main/java/com/starrocks/sql/analyzer/SetStmtAnalyzer.java

```diff
@@ -73,7 +73,16 @@ public static void analyze(SetStmt setStmt, ConnectContext session) {
             if (var instanceof SystemVariable) {
                 analyzeSystemVariable((SystemVariable) var);
             } else if (var instanceof UserVariable) {
-                analyzeUserVariable((UserVariable) var);
+                UserVariable userVar = (UserVariable) var;
+                UserVariable userVariableExisted = session.getUserVariable(userVar.getVariable());
+                if (userVariableExisted != null) {
+                    session.removeUserVariable(userVar.getVariable());
```

Comment:
It is not acceptable to remove something from a session in analyze status. 
It will change the session no matter whether the statement executes successfully or not.
And you need to add more comments in the code to explain why you did it this way.